pkg/scrapers/twitter/tweets.go

```diff
@@ -3,212 +3,184 @@ package twitter
-//
-// Returns:
-//   - A string representing the sentiment analysis prompt.
-//   - A string representing the sentiment analysis result.
-//   - An error if the scraping or sentiment analysis process encounters any issues.
-func ScrapeTweetsForSentiment(query string, count int, model string) (string, string, error) {
-	scraper := auth()
-	var tweets []*TweetResult
-
-	if scraper == nil {
-		return "", "", fmt.Errorf("there was an error authenticating with your Twitter credentials")
-	}
-
-	// Set search mode
-	scraper.SetSearchMode(twitterscraper.SearchLatest)
-
-	// Perform the search with the specified query and count
-	for tweetResult := range scraper.SearchTweets(context.Background(), query, count) {
-		var tweet TweetResult
-		if tweetResult.Error != nil {
-			tweet = TweetResult{
-				Tweet: nil,
-				Error: tweetResult.Error,
-			}
-		} else {
-			tweet = TweetResult{
-				Tweet: &tweetResult.Tweet,
-				Error: nil,
-			}
+	for _, pair := range accountPairs {
+		credentials := strings.Split(pair, ":")
+		if len(credentials) != 2 {
+			logrus.Warnf("invalid account credentials: %s", pair)
+			continue
 		}
-		tweets = append(tweets, &tweet)
-	}
-	sentimentPrompt := "Please perform a sentiment analysis on the following tweets, using an unbiased approach. Sentiment analysis involves identifying and categorizing opinions expressed in text, particularly to determine whether the writer's attitude towards a particular topic, product, etc., is positive, negative, or neutral. After analyzing, please provide a summary of the overall sentiment expressed in these tweets, including the proportion of positive, negative, and neutral sentiments if applicable."
-
-	twitterScraperTweets := make([]*twitterscraper.TweetResult, len(tweets))
-	for i, tweet := range tweets {
-		twitterScraperTweets[i] = &twitterscraper.TweetResult{
-			Tweet: *tweet.Tweet,
-			Error: tweet.Error,
+		account := &TwitterAccount{
+			Username: strings.TrimSpace(credentials[0]),
+			Password: strings.TrimSpace(credentials[1]),
 		}
+		accounts = append(accounts, account)
 	}
-	prompt, sentiment, err := llmbridge.AnalyzeSentimentTweets(twitterScraperTweets, model, sentimentPrompt)
-	if err != nil {
-		return "", "", err
-	}
-	return prompt, sentiment, tweets[0].Error
+
+	return accounts
 }
 
-// ScrapeTweetsByQuery performs a search on Twitter for tweets matching the specified query.
-// It fetches up to the specified count of tweets and returns a slice of Tweet pointers.
-// Parameters:
-//   - query: The search query string to find matching tweets.
-//   - count: The maximum number of tweets to retrieve.
-//
-// Returns:
-//   - A slice of pointers to twitterscraper.Tweet objects that match the search query.
-//   - An error if the scraping process encounters any issues.
 func ScrapeTweetsByQuery(query string, count int) ([]*TweetResult, error) {
-	scraper := auth()
-	var tweets []*TweetResult
-	var lastError error
+	once.Do(initializeAccountManager)
```

Comment:
if we want to really go with a singleton, I'd push back `once.Do` in favor of simply using `init()` - in this case we really know where it gets created